test/output/callbacks/mfg_event_converter_test.py

```diff
@@ -0,0 +1,352 @@
+    self.assertEqual(mock_measurement.name, u'mock-measurement-name')
+    self.assertEqual(mock_measurement.name, u'mock-measurement-name')
+
+  def testCopyAttachmentsFromPhase(self):
+    attachment = test_record.Attachment('mock-data', 'text/plain')
+    phase = test_record.PhaseRecord(
+        name='mock-phase-name',
+        descriptor_id=1,
+        codeinfo=self.create_codeinfo(),
+        attachments={'mock-attachment-name': attachment},
+    )
+
+    mfg_event = mfg_event_pb2.MfgEvent()
+    copier = mfg_event_converter.PhaseCopier([phase])
+    copier.copy_attachments(mfg_event)
+
+    self.assertEqual(mfg_event.attachment[0].name, 'mock-attachment-name')
+    self.assertEqual(mfg_event.attachment[0].value_binary, b'mock-data')
+    self.assertEqual(mfg_event.attachment[0].type, test_runs_pb2.TEXT_UTF8)
+
+
+class MultiDimConversionTest(unittest.TestCase):
+
+  @classmethod
+  def create_multi_dim_measurement(cls):
+    """Util function to create a test multi-dim measurement."""
+    measurement = measurements.Measurement('test_measurement')
+    measurement.with_units('°C').with_dimensions('ms', 'assembly', 'zone')
+    for t in range(10):
+      for assembly in ['A', 'B', 'C']:
+        for zone in range(3):
+          temp = zone + t
+          dims = (t, assembly, zone)
+          measurement.measured_value[dims] = temp
+
+    measurement.outcome = measurements.Outcome.PASS
+
+    return measurement
+
+  def test_multidim_measurement_to_attachment(self):
+    """Test for the full conversion flow."""
+    meas = self.create_multi_dim_measurement()
+
+    attachment = mfg_event_converter.multidim_measurement_to_attachment(
+        name='test_measurement_multidim', measurement=meas)
+
+    self.assertEqual(
+        json.loads(TEST_MULTIDIM_JSON), json.loads(attachment.data))
+
+  def test_attachment_to_multidim_measurement(self):
+    expected = self.create_multi_dim_measurement()
+
+    attachment = test_record.Attachment(TEST_MULTIDIM_JSON,
+                                        test_runs_pb2.MULTIDIM_JSON)
+    measurement = mfg_event_converter.attachment_to_multidim_measurement(
+        attachment)
+
+    self.assertEqual(expected.measured_value.value,
+                     measurement.measured_value.value)
+    for exp, act in zip(expected.dimensions, measurement.dimensions):
+      self.assertEqual(exp, act)
+
+  def test_reversibleish(self):
+    """Veriy that in happy case multidim -> attachment is reversible."""
+    mdim = self.create_multi_dim_measurement()
+
+    attachment = mfg_event_converter.multidim_measurement_to_attachment(
+        name='test_measurement_multidim', measurement=mdim)
+
+    reversed_mdim = mfg_event_converter.attachment_to_multidim_measurement(
+        attachment)
+
+    self.assert_same_mdim(mdim, reversed_mdim)
+
+  def test_reversibleish_leagcy_status_int(self):
+    """Veriy that in happy case multidim -> attachment is reversible."""
```

Comment:
Maybe mention the intermediate int casting of the outcome field?

Reply:
Done.